### PR TITLE
Update location of `resolve-href` in next 13.5.2

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: https://github.com/blitz-js/blitz/issues/4221

### What are the changes and their implications?

- [x] upgrade local packages to latest next
- [x] update location of `resolve-href` to  `next/dist/client/resolve-href`
- [x] upgrade workflows to node 18 

## Bug Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)

### CODE REVIEW

**Review by Kosai106 — APPROVED** [review]
Yes please 🙏  I just encountered this issue myself.

File: integration-tests/utils/next-test-utils.ts [hunk 1]
```diff
@@ -435,9 +436,9 @@ export function runNextCommandDev(argv, stdOut, opts: RunNextCommandDevOptions =
         opts.onStdout(message)
       }
 
-      if (opts.stdout !== false) {
-        process.stdout.write(message)
-      }
+      // if (opts.stdout !== false) {
+      process.stdout.write(message)
+      // }
```
Comment (siddhsuresh) [inline integration-tests/utils/next-test-utils.ts:441]:
```suggestion
      if (opts.stdout !== false) {
      	process.stdout.write(message)
      }
```

File: .changeset/lucky-teachers-sleep.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+"@blitzjs/auth": patch
+"@blitzjs/next": patch
+"@blitzjs/rpc": patch
+"blitz": patch
+---
+
+Fix `Error: Cannot find module 'next/dist/shared/lib/router/utils/resolve-href'` by updating the location of next.js internal function.
```
Comment (siddhsuresh) [inline .changeset/lucky-teachers-sleep.md:8]:
```suggestion
---
"@blitzjs/auth": patch
"@blitzjs/next": major
"@blitzjs/rpc": patch
"blitz": patch
---
⚠️ Breaking Change:
Next.js version 13.5 or above is now required to use `@blitzjs/next`

Fix `Error: Cannot find module 'next/dist/shared/lib/router/utils/resolve-href'` by updating the location of next.js internal function.
```

**Comment by bravo-kernel** [comment]
ty 👍

**Pull request merged**